apps/src/lab2/levelEditors/aichatSettings/ModelSelectionFields.tsx

```diff
@@ -0,0 +1,107 @@
+import React, {useContext, useState, useMemo, useCallback} from 'react';
+
+import {BodyFourText} from '@cdo/apps/componentLibrary/typography';
+import Checkbox from '@cdo/apps/componentLibrary/checkbox/Checkbox';
+import SimpleDropdown from '@cdo/apps/componentLibrary/dropdown/simpleDropdown';
+
+import {modelDescriptions} from '@cdo/apps/aichat/constants';
+import moduleStyles from './edit-aichat-settings.module.scss';
+import {UpdateContext} from './UpdateContext';
+import FieldSection from './FieldSection';
+import {Visibility} from '@cdo/apps/aichat/types';
+
+const ModelSelectionFields: React.FunctionComponent = () => {
+  const {setModelSelectionValues, aichatSettings} = useContext(UpdateContext);
+  const shouldDisableAdditionalModelSelection =
+    aichatSettings.visibilities.selectedModelId !== Visibility.EDITABLE;
+  const [additionalAvailableModelIds, setAdditionalAvailableModelIds] =
+    useState<Set<string>>(
+      new Set(
+        aichatSettings.availableModelIds.filter(
+          id => id !== aichatSettings.initialCustomizations.selectedModelId
+        )
+      )
+    );
+  const selectedModelId = aichatSettings.initialCustomizations.selectedModelId;
+
+  const modelDropdownItems = useMemo(() => {
```

Comment:
hah I also got twisted up in React hooks and at second glance I think this could just be a const outside of the React component, it can't really change yeah?

Reply:
Yeah, this shouldn't ever change since the only dependency is the modelDescriptions which is imported from consts

Reply:
OK, moved outside the component!